codalab/lib/bundle_cli.py

```diff
@@ -694,13 +694,16 @@ def do_edit_command(self, argv, parser):
 
         metadata = info['metadata']
         new_metadata = copy.deepcopy(metadata)
+        is_new_metadata_updated = False
         if args.name:
             new_metadata['name'] = args.name
+            is_new_metadata_updated = True
         if args.description:
             new_metadata['description'] = args.description
+            is_new_metadata_updated = True
 
         # Prompt user for all information
-        if not self.headless and metadata == new_metadata:
+        if not is_new_metadata_updated and not self.headless and metadata == new_metadata:
```

Comment:
is_new_metadata_updated serves like a flag. Whenever anything in the new_metadata is changed, it is set to True, thus avoiding opening the editor.
 
If this variable were absent, then in the use case where old and the new value are the same, the editor would open (even when the new value was passed with cl edit command)